From a2cbc1a9cf6df0ebdcb0463b68e519c6d8b141be Mon Sep 17 00:00:00 2001
From: Marcel Ziswiler <marcel.ziswiler@toradex.com>
Date: Thu, 23 Nov 2017 12:57:07 +0100
Subject: [PATCH] apalis-tk1: tezi specific kernel configuration

Toradex Easy Installer specific Linux kernel configuration incl.
built-in USB 3.0 XUSB firmware (needs deploying separately).

Signed-off-by: Marcel Ziswiler <marcel.ziswiler@toradex.com>
---
 arch/arm/configs/apalis-tk1_defconfig | 75 ++++-------------------------------
 1 file changed, 8 insertions(+), 67 deletions(-)

diff --git a/arch/arm/configs/apalis-tk1_defconfig b/arch/arm/configs/apalis-tk1_defconfig
index da141c0b41aa..9d52989741e4 100644
--- a/arch/arm/configs/apalis-tk1_defconfig
+++ b/arch/arm/configs/apalis-tk1_defconfig
@@ -89,7 +89,6 @@ CONFIG_INET6_AH=y
 CONFIG_INET6_ESP=y
 CONFIG_INET6_IPCOMP=y
 CONFIG_IPV6_MIP6=y
-CONFIG_IPV6_SIT=m
 CONFIG_IPV6_TUNNEL=y
 CONFIG_IPV6_MULTIPLE_TABLES=y
 CONFIG_NETFILTER=y
@@ -145,29 +144,15 @@ CONFIG_NET_CLS_ACT=y
 CONFIG_NET_ACT_POLICE=y
 CONFIG_NET_ACT_GACT=y
 CONFIG_NET_ACT_MIRRED=y
-CONFIG_CAN=y
-CONFIG_CAN_MCP251X=y
-CONFIG_CAN_APALIS_TK1_K20=m
-CONFIG_BT=m
-CONFIG_BT_RFCOMM=m
-CONFIG_BT_BNEP=m
-CONFIG_BT_HIDP=m
-CONFIG_BT_HCIBTUSB=m
-CONFIG_CFG80211=m
-CONFIG_NL80211_TESTMODE=y
-CONFIG_CFG80211_WEXT=y
-CONFIG_MAC80211=m
-CONFIG_MAC80211_LEDS=y
-CONFIG_RFKILL=y
-CONFIG_RFKILL_INPUT=y
-CONFIG_RFKILL_GPIO=y
-CONFIG_CAIF=y
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
-# CONFIG_FIRMWARE_IN_KERNEL is not set
+CONFIG_EXTRA_FIRMWARE="tegra_xusb_firmware"
+# CONFIG_FW_LOADER_USER_HELPER is not set
 CONFIG_CMA=y
 CONFIG_PLATFORM_ENABLE_IOMMU=y
 CONFIG_BLK_DEV_LOOP=y
+CONFIG_BLK_DEV_RAM=y
+CONFIG_BLK_DEV_RAM_SIZE=65536
 CONFIG_SENSORS_NCT1008=y
 CONFIG_TEGRA_CRYPTO_DEV=y
 CONFIG_THERM_EST=y
@@ -179,40 +164,25 @@ CONFIG_BLK_DEV_SR_VENDOR=y
 CONFIG_CHR_DEV_SG=y
 CONFIG_SCSI_MULTI_LUN=y
 CONFIG_ATA=y
-CONFIG_SATA_AHCI=m
+CONFIG_SATA_AHCI=y
 CONFIG_SATA_AHCI_TEGRA=y
 CONFIG_NETDEVICES=y
 CONFIG_DUMMY=y
 CONFIG_TUN=y
-CONFIG_E1000E=m
 CONFIG_IGB=y
-CONFIG_PPP=m
-CONFIG_PPP_BSDCOMP=m
-CONFIG_PPP_DEFLATE=m
-CONFIG_PPP_FILTER=y
-CONFIG_PPP_MPPE=m
-CONFIG_PPPOLAC=m
-CONFIG_PPPOPNS=m
-CONFIG_PPP_ASYNC=m
-CONFIG_PPP_SYNC_TTY=m
 CONFIG_USB_USBNET=y
 # CONFIG_USB_NET_CDC_NCM is not set
 # CONFIG_USB_NET_NET1080 is not set
 # CONFIG_USB_BELKIN is not set
 # CONFIG_USB_NET_ZAURUS is not set
-CONFIG_INPUT_POLLDEV=y
 # CONFIG_INPUT_MOUSEDEV is not set
-CONFIG_INPUT_JOYDEV=m
 CONFIG_INPUT_EVDEV=y
 CONFIG_INPUT_KEYRESET=y
 CONFIG_INPUT_CFBOOST=y
 # CONFIG_KEYBOARD_ATKBD is not set
 CONFIG_KEYBOARD_GPIO=y
 # CONFIG_INPUT_MOUSE is not set
-CONFIG_INPUT_JOYSTICK=y
-CONFIG_JOYSTICK_XPAD=m
 CONFIG_INPUT_TOUCHSCREEN=y
-CONFIG_TOUCHSCREEN_APALIS_TK1_K20=m
 CONFIG_INPUT_MISC=y
 CONFIG_INPUT_KEYCHORD=y
 CONFIG_INPUT_UINPUT=y
@@ -223,7 +193,6 @@ CONFIG_DEVPTS_MULTIPLE_INSTANCES=y
 # CONFIG_DEVKMEM is not set
 CONFIG_SERIAL_8250=y
 CONFIG_SERIAL_8250_CONSOLE=y
-CONFIG_SERIAL_8250_PCI=m
 CONFIG_SERIAL_8250_NR_UARTS=9
 CONFIG_SERIAL_8250_RUNTIME_UARTS=9
 CONFIG_SERIAL_TEGRA=y
@@ -238,7 +207,6 @@ CONFIG_SPI_SPIDEV=y
 CONFIG_PINCTRL_AS3722=y
 CONFIG_DEBUG_GPIO=y
 CONFIG_GPIO_SYSFS=y
-CONFIG_GPIO_APALIS_TK1_K20=m
 CONFIG_POWER_RESET_AS3722=y
 CONFIG_THERMAL_GOV_PID=y
 CONFIG_GENERIC_ADC_THERMAL=y
@@ -249,10 +217,6 @@ CONFIG_WATCHDOG_CORE=y
 CONFIG_WATCHDOG_NOWAYOUT=y
 CONFIG_TEGRA_WATCHDOG=y
 CONFIG_TEGRA_WATCHDOG_ENABLE_ON_PROBE=y
-CONFIG_SSB=m
-CONFIG_SSB_DRIVER_PCICORE=y
-CONFIG_MFD_APALIS_TK1_K20=m
-CONFIG_APALIS_TK1_K20_EZP=y
 CONFIG_MFD_AS3722=y
 CONFIG_REGULATOR=y
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
@@ -271,20 +235,10 @@ CONFIG_V4L_PLATFORM_DRIVERS=y
 CONFIG_TEGRA_NVAVP=y
 CONFIG_TEGRA_NVAVP_AUDIO=y
 CONFIG_SOC_CAMERA=y
-CONFIG_SOC_CAMERA_PLATFORM=m
-CONFIG_VIDEO_TEGRA=m
-CONFIG_SOC_CAMERA_AR0261=m
-CONFIG_SOC_CAMERA_AR0330=m
-CONFIG_SOC_CAMERA_AP1302=m
-CONFIG_SOC_CAMERA_IMX135=m
-CONFIG_SOC_CAMERA_OV5640=m
-CONFIG_SOC_CAMERA_TC358743=m
-CONFIG_SOC_CAMERA_ADV7280=m
 # CONFIG_VGA_ARB is not set
 CONFIG_VIDEO_OUTPUT_CONTROL=y
 CONFIG_FB=y
 CONFIG_TEGRA_GRHOST=y
-CONFIG_TEGRA_GRHOST_VI=m
 CONFIG_TEGRA_DC=y
 CONFIG_TEGRA_DP=y
 CONFIG_TEGRA_LVDS=y
@@ -318,7 +272,7 @@ CONFIG_SND_SOC=y
 CONFIG_SND_SOC_TEGRA=y
 CONFIG_SND_SOC_TEGRA_APALIS_TK1=y
 CONFIG_HIDRAW=y
-CONFIG_HID_MULTITOUCH=m
+CONFIG_HID_MULTITOUCH=y
 CONFIG_USB_HIDDEV=y
 CONFIG_I2C_HID=y
 CONFIG_USB=y
@@ -328,20 +282,12 @@ CONFIG_USB_OTG=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_TEGRA_XUSB_PLATFORM=y
 CONFIG_USB_EHCI_HCD=y
-CONFIG_USB_ACM=m
-CONFIG_USB_WDM=m
 CONFIG_USB_STORAGE=y
 CONFIG_USB_SERIAL=y
 CONFIG_USB_SERIAL_CONSOLE=y
 CONFIG_USB_SERIAL_GENERIC=y
 CONFIG_USB_SERIAL_PL2303=y
-CONFIG_USB_SERIAL_OPTION=m
-CONFIG_USB_SERIAL_BASEBAND=m
 CONFIG_USB_TEGRA_OTG=y
-CONFIG_USB_GADGET=y
-CONFIG_USB_GADGET_VBUS_DRAW=500
-CONFIG_USB_TEGRA=y
-CONFIG_USB_G_ANDROID=y
 CONFIG_MMC=y
 CONFIG_MMC_UNSAFE_RESUME=y
 CONFIG_MMC_BLOCK_MINORS=16
@@ -354,8 +300,6 @@ CONFIG_NEW_LEDS=y
 CONFIG_LEDS_CLASS=y
 CONFIG_LEDS_GPIO=y
 CONFIG_LEDS_PWM=y
-CONFIG_LEDS_TRIGGER_TIMER=y
-CONFIG_LEDS_TRIGGER_DEFAULT_ON=y
 CONFIG_SWITCH=y
 CONFIG_RTC_CLASS=y
 CONFIG_RTC_DRV_AS3722=y
@@ -373,7 +317,6 @@ CONFIG_TEGRA_IOMMU_SMMU=y
 CONFIG_PM_DEVFREQ=y
 CONFIG_EXTCON=y
 CONFIG_IIO=y
-CONFIG_APALIS_TK1_K20_ADC=m
 CONFIG_PWM=y
 CONFIG_PWM_TEGRA=y
 CONFIG_GK20A_PMU=y
@@ -385,11 +328,12 @@ CONFIG_EXT4_FS_SECURITY=y
 CONFIG_AUTOFS4_FS=y
 CONFIG_FUSE_FS=y
 CONFIG_CUSE=y
+CONFIG_UDF_FS=y
 CONFIG_VFAT_FS=y
 CONFIG_NTFS_FS=y
 CONFIG_TMPFS=y
 CONFIG_TMPFS_POSIX_ACL=y
-CONFIG_SQUASHFS=m
+CONFIG_SQUASHFS=y
 CONFIG_SQUASHFS_XATTR=y
 CONFIG_SQUASHFS_LZO=y
 CONFIG_SQUASHFS_XZ=y
@@ -409,7 +353,6 @@ CONFIG_NLS_DEFAULT="cp437"
 CONFIG_NLS_CODEPAGE_437=y
 CONFIG_NLS_ASCII=y
 CONFIG_NLS_ISO8859_1=y
-CONFIG_NLS_ISO8859_15=m
 CONFIG_NLS_UTF8=y
 CONFIG_PRINTK_TIME=y
 CONFIG_DEBUG_SECTION_MISMATCH=y
@@ -428,10 +371,8 @@ CONFIG_FUNCTION_TRACER=y
 # CONFIG_FUNCTION_GRAPH_TRACER is not set
 CONFIG_DYNAMIC_DEBUG=y
 CONFIG_DEBUG_USER=y
-CONFIG_CRYPTO_TEST=m
 CONFIG_CRYPTO_CCM=y
 CONFIG_CRYPTO_GCM=y
 CONFIG_CRYPTO_CMAC=y
-CONFIG_CRYPTO_TWOFISH=m
 # CONFIG_CRYPTO_ANSI_CPRNG is not set
 CONFIG_CRYPTO_DEV_TEGRA_SE=y
-- 
2.16.2

